src/6_branch.js

```diff
@@ -334,13 +352,17 @@ Branch.prototype['init'] = wrap(
 				resources.hasApp,
 				{ "browser_fingerprint_id": currentSessionData['browser_fingerprint_id'] },
 				function(err, has_app) {
-					if (has_app && !currentSessionData['has_app']) {
+					if (err) {
+						self.init_state_fail_code = init_state_fail_codes.HAS_APP_FAILED;
+						self.init_state_fail_details = err.message;
+					}
+					if (!err && has_app && !currentSessionData['has_app']) {
 						currentSessionData['has_app'] = true;
 						session.update(self._storage, currentSessionData);
 						self._publishEvent('didDownloadApp');
 					}
 					if (cb) {
-						cb(err, currentSessionData);
+						cb(null, currentSessionData);
```

Comment:
This particularly had potential for failure.  It seemed like an outside chance, but patching it just in case.